core/src/main/java/tech/pegasys/eth2signer/core/http/handlers/SignForPublicKeyHandler.java

```diff
@@ -51,11 +60,23 @@ public void handle(RoutingContext routingContext) {
     }
 
     final Bytes dataToSign = getDataToSign(params);
-    final BLSSignature signature = signer.get().sign(dataToSign);
+    final ArtifactSignature artifactSignature = signer.get().sign(dataToSign);
+    final String formatSignature = formatSignature(artifactSignature);
+
     routingContext
         .response()
         .putHeader(CONTENT_TYPE, PLAIN_TEXT_UTF_8.toString())
-        .end(signature.toString());
+        .end(formatSignature);
+  }
+
+  @SuppressWarnings("unchecked")
+  private String formatSignature(final ArtifactSignature signature) {
+    if (signature.getType() == type) {
+      final T artifactSignature = (T) signature;
+      return signatureFormatter.format(artifactSignature);
+    } else {
+      throw new IllegalStateException("Invalid signature type");
```

Comment:
if we hit this exception - what does it mean for the user (and how does it present?)? That the key they tried to sign with, wasn't a BLS key? So ... an INTERNAL_SERVER_ERROR?

Reply:
This will return an internal server error. This case shouldn't happen and represents an internal configuration error with how we have wired our application together